Source/WebKit/WebKitSwift/TextAnimation/TextAnimationManager.swift

```diff
@@ -114,8 +114,18 @@ extension TextAnimationManager: UITextEffectView.ReplacementTextEffect.Delegate
     
     public func replacementEffectDidComplete(_ effect: UITextEffectView.ReplacementTextEffect) {
         self.effectView.removeEffect(effect.id)
-        
-        // FIXME: remove the effect from the chunkToEffect map rdar://126307144
+
+        for (animationID, _) in chunkToEffect {
+            if (chunkToEffect[animationID] == effect.id) {
+                chunkToEffect.removeValue(forKey: animationID)
+
+                guard let delegate = self.delegate else {
+                    Self.logger.debug("Can't obtain Targeted Preview. Missing delegate." )
```

Comment:
Nit - extra space before the `)`

```suggestion
                    Self.logger.debug("Can't obtain Targeted Preview. Missing delegate.")
```